packages/tests/integration/src/SingleTableInheritance.test.ts

```diff
@@ -390,4 +372,40 @@ describe("SingleTableInheritance", () => {
     expect(ot.transientFields.oldReactiveRuleRan).toBe(true);
     expect(nt.transientFields.newReactiveRuleRan).toBe(true);
   });
+
+  it("setDefaults work as expected for subtypes", async () => {
+    const em = newEntityManager();
+    const ot = newTaskOld(em, {});
+    const nt = newTaskNew(em, {});
+    await em.flush();
+    // Then TaskOld runs its defaults
+    expect(ot).toMatchEntity({
+      syncDefault: "TaskOld",
+      asyncDefault_1: "TaskOld Async1",
+      asyncDefault_2: "TaskOld Async1 Async2",
+    });
+    // And TaskNew runs its defaults
+    expect(nt).toMatchEntity({
+      syncDefault: "TaskNew",
+      asyncDefault_1: "TaskNew Async1",
+      asyncDefault_2: "TaskNew Async1 Async2",
+    });
+  });
+
+  it("derived fields work as expected", async () => {
```

Comment:
NOTE: Despite these tests working here, I'm still experiencing issues with this in graphql-service. I want to ship this as-is, and try to debug the graphql-service use case separately.